plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/impls/materialize/MaterializeSchemaCache.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.postgresql.model.impls.materialize;
+
+import java.sql.SQLException;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.ext.postgresql.model.PostgreDatabase;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCSession;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+
+public class MaterializeSchemaCache extends PostgreDatabase.SchemaCache {
+    @Override
+    protected MaterializeSchema fetchObject(@NotNull JDBCSession session, @NotNull PostgreDatabase owner,
+            @NotNull JDBCResultSet resultSet) throws SQLException {
+        String name = JDBCUtils.safeGetString(resultSet, "nspname");
+        if (name == null) {
+            return null;
+        }
+        if (MaterializeSchema.isUtilitySchema(name) && 
```

Comment:
Do we really need this check? Schema won't be shown in UI if isShowUtilityObjects==false and schema is utility one

Reply:
Good catch, thank you for pointing that out! I've just tested it and verified that this is indeed not needed. Just updated the PR accordingly. Let me know if anything else pops up.

Reply:
I believe the question was about the `MaterializeSchema.isUtilitySchema(name)` expression, not the one above.

Reply:
The logic was actually taken from an existing driver which has the same structure:

> https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.greenplum/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumSchemaCache.java#L37-L41

If you believe that this is not needed, I can go ahead and remove it.

Reply:
I've just tested this and indeed it works fine without that extra expression. I've gone ahead and removed it! Let me know what you think.